### PR TITLE
CLI commands for AzureDiskEncryption

### PR DESCRIPTION
**Pull request by SudhakaraReddyEvuri**
This change list contains CLI commands for AzureDiskEncryption feature. Two new commands are added.
1. azure vm enable-disk-encryption
2. azure vm show-disk-encryption-status

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@SudhakaraReddyEvuri__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -1169,7 +1169,186 @@ __.extend(VMClient.prototype, {
...
+    // Get extension status from VM instance view
+	var extensionName = '';
+	var publisherName = '';
+	
+    if (utils.ignoreCaseEquals(params.osType, 'Linux')) {
+	    extensionName = vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_LINUX_EXTENSION_NAME;
+		publisherName = vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_LINUX_EXTENSION_PUBLISHER;
+    }
+    else if(utils.ignoreCaseEquals(params.osType, 'Windows')) {
+	    extensionName = vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_WINDOWS_EXTENSION_NAME;
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: please fix the indent

File: lib/commands/arm/vm/vm._js [hunk 2]
```diff
@@ -378,6 +391,44 @@ exports.init = function (cli) {
...
+    .option('-k, --disk-encryption-key-vault-url <disk-encryption-key-vault-url>', $('URL of the KeyVault where generated encryption key will be placed to'))
+    .option('-r, --disk-encryption-key-vault-id <disk-encryption-key-vault-id>', $('ResourceID of the KeyVault where generated encryption key will be placed to'))
+    .option('-u, --key-encryption-key-url <key-encryption-key-url>', $('Versioned KeyVault URL of the KeyEncryptionKey used to encrypt the disk encryption key'))
+    .option('-i, --key-encryption-key-vault-id <key-encryption-key-vault-id>', $('ResourceID of the KeyVault containing the KeyEncryptionKey used to encrypt the disk encryption key'))
+    .option('-l, --key-encryption-algorithm <key-encryption-algorithm>', $('KeyEncryption Algorithm used to encrypt the volume encryption key'))
+    .option('-t, --volume-type <volume-type>', $('Type of the volume OS or Data or All to perform encryption operation'))
+    .option('-x, --sequence-version <sequence-version>', $('Sequence version of encryption operation. This must be incremented to perform repeated encryption operations on the same VM'))
+	.option('-q, --quiet', $('quiet mode, do not ask for confirmation'))
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vm._js]: indent

File: lib/commands/arm/vm/vmClient._js [hunk 3]
```diff
@@ -1169,7 +1169,186 @@ __.extend(VMClient.prototype, {
...
+    var subscription = profile.current.getSubscription(this.subscription);
+    var serviceClients = this._getServiceClients(subscription);
+    var virtualMachine = new VirtualMachine(this.cli, serviceClients);
+    var output = this.cli.output;
+    var encryptionStatus = {
+      osVolumeEncrypted: false,
+      osVolumeEncryptionSettings: null,
+      dataVolumesEncrypted: false
+    };
+	var extensionStatus = null;
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: indent here

File: lib/commands/arm/vm/vmClient._js [hunk 4]
```diff
@@ -1169,7 +1169,186 @@ __.extend(VMClient.prototype, {
...
+
+    if(vmModel.virtualMachine.storageProfile.oSDisk.encryptionSettings)
+    {
+      encryptionStatus.osVolumeEncryptionSettings = vmModel.virtualMachine.storageProfile.oSDisk.encryptionSettings;
+      encryptionStatus.osVolumeEncrypted = true;
+
+      // Data volume encryption is considered enabled only when OS volume encryption is enabled
+      // Get extension model view
+	  
+	  var extensionName = '';
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: and here

**Comment by yugangw-msft** [comment]
Since the PR have no scenario tests, please do manual tests to verify your cmdlet work. We are late alreayd so would be very difficult for us to take another patch

File: lib/commands/arm/vm/vmClient._js (hunk 1 above)
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vmClient._js]: Fixed it

File: lib/commands/arm/vm/vm._js (hunk 2 above)
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vm._js]: Fixed it

File: lib/commands/arm/vm/vmClient._js (hunk 4 above)
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vmClient._js]: Fixed it

File: lib/commands/arm/vm/vm._js (hunk 2 above)
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vm._js]: Fixed it

**Comment by huangpf** [comment]
@yugangw-msft, Could you please take this one first? https://github.com/Azure/azure-xplat-cli/pull/2243

Sudhakar's PR contains some of my commits that were not submitted... Thanks.

**Comment by huangpf** [comment]
LGTM.

**Pull request merged**